src/main/java/com/sngular/kloadgen/randomtool/random/RandomObject.java

```diff
@@ -174,23 +183,26 @@ private BigDecimal getDecimalValueOrRandom(final Integer valueLength, final List
       value = new BigDecimal(fieldValueList.get(RandomUtils.nextInt(0, fieldValueList.size())).trim());
     } else {
       final Number minimum = calculateMinimum(valueLength - 1, constraints);
-      Number maximum = calculateMaximum(valueLength - 1, constraints);
+
+      Number maximum;
+      maximum = calculateMaximum(valueLength - 1, constraints);
 
       if (constraints.containsKey(ConstraintTypeEnum.MULTIPLE_OF)) {
         final int multipleOf = Integer.parseInt(constraints.get(ConstraintTypeEnum.MULTIPLE_OF));
         maximum = maximum.intValue() > multipleOf ? maximum.intValue() / multipleOf : maximum;
         value = BigDecimal.valueOf(RandomUtils.nextDouble(minimum.doubleValue(), maximum.doubleValue()) * multipleOf);
       } else {
-        if (valueLength < 3) {
-          value = new BigDecimal(getIntegerValueOrRandom(valueLength, fieldValueList, constraints));
-        } else {
-          final BigDecimal aux = BigDecimal.valueOf(RandomUtils.nextLong(minimum.longValue(), maximum.longValue()));
-          final int decLength = RandomUtils.nextInt(1, valueLength / 2);
-          value = aux.multiply(BigDecimal.valueOf(0.1).pow(decLength));
-        }
+/*                if (valueLength < 3) {
```

Comment:
Please remove this comment code if not used anymore

Reply:
listo